libdnf5/repo/solv_repo.cpp

```diff
@@ -792,4 +792,9 @@ void SolvRepo::create_environment_solvable(
     repodata_internalize(data);
 }
 
+void SolvRepo::empty() {
+    // TODO(mblaha): is resuseid=true safe enough?
```

Comment:
Code from libsolv:
```
  if (reuseids && repo->end == pool->nsolvables)
    {
      /* it's ok to reuse the ids. As this is the last repo, we can
         just shrink the solvable array */
```
It looks like `reuseids` works only when the repo is last. DNF places the system repo first. Behind it are available repos. The system repo is last only if it is the only one (we don't have available repos loaded).